lib/routes/withdraw/reverse_swap/confirmation_page/reverse_swap_confirmation_page.dart

```diff
@@ -57,44 +52,61 @@ class _ReverseSwapConfirmationPageState extends State<ReverseSwapConfirmationPag
         future: _fetchFeeOptionsFuture,
         builder: (context, snapshot) {
           if (snapshot.error != null) {
-            return _ErrorMessage(
-              message: texts.reverse_swap_confirmation_error_fetch_fee,
-            );
+            return _ErrorMessage(message: texts.reverse_swap_confirmation_error_fetch_fee);
           }
           if (snapshot.connectionState != ConnectionState.done) {
             return const Center(child: Loader());
           }
 
           if (affordableFees.isNotEmpty) {
             return FeeChooser(
-              walletBalance: widget.amountSat,
+              amountSat: widget.amountSat,
               feeOptions: snapshot.data!,
               selectedFeeIndex: selectedFeeIndex,
-              boltzFees: widget.boltzFees,
               onSelect: (index) => setState(() {
                 selectedFeeIndex = index;
               }),
+              isMaxValue: widget.isMaxValue,
             );
           } else {
-            return _ErrorMessage(
-              message: texts.reverse_swap_confirmation_error_funds_fee,
-            );
+            return _ErrorMessage(message: texts.reverse_swap_confirmation_error_funds_fee);
           }
         },
       ),
       bottomNavigationBar:
           (affordableFees.isNotEmpty && selectedFeeIndex >= 0 && selectedFeeIndex < affordableFees.length)
               ? SafeArea(
                   child: ReverseSwapButton(
-                    amountSat: widget.amountSat,
-                    onchainRecipientAddress: widget.onchainRecipientAddress,
-                    satPerVbyte: affordableFees[selectedFeeIndex].feeVByte,
-                    feesHash: widget.feesHash,
+                    recipientAddress: widget.onchainRecipientAddress,
+                    prepareOnchainPaymentResponse: selectedFeeOption.pairInfo,
                   ),
                 )
               : null,
     );
   }
+
+  void _fetchReverseSwapFeeOptions() {
+    final reverseSwapBloc = context.read<ReverseSwapBloc>();
+    _fetchFeeOptionsFuture = reverseSwapBloc.fetchReverseSwapFeeOptions(
+      amountSat: widget.amountSat,
+      amountType: widget.isMaxValue ? SwapAmountType.Receive : SwapAmountType.Send,
```

Comment:
Shouldn't this be the other way around? I mean other than if the user wants to drain all the funds they want to input the recipient amount. Isn't it the way in breezmobile?